# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieChangeEvent.h
# Commit: cc3c39aa8fb8
# Full Hash: cc3c39aa8fb8124cca265e84f71a29cbd218b704
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r 2fc9a3d0ad8b -r cc3c39aa8fb8 dom/cookiestore/CookieChangeEvent.h
--- a/dom/cookiestore/CookieChangeEvent.h	Wed Sep 11 16:26:54 2024 +0000
+++ b/dom/cookiestore/CookieChangeEvent.h	Wed Sep 11 16:26:55 2024 +0000
@@ -29,6 +29,12 @@
 
   void GetDeleted(nsTArray<CookieListItem>& aList) const;
 
+  static already_AddRefed<CookieChangeEvent> CreateForChangedCookie(
+      EventTarget* aEventTarget, const CookieListItem& aItem);
+
+  static already_AddRefed<CookieChangeEvent> CreateForDeletedCookie(
+      EventTarget* aEventTarget, const CookieListItem& aItem);
+
  private:
   ~CookieChangeEvent();
 